cli/src/main.rs

```diff
@@ -1,235 +1,162 @@
+    supports_color::on(supports_color::Stream::Stdout).is_some()
+}
 
-    if env::args().any(|a| VERSION_ARG.contains(&a.as_str())) {
-        print_version(&styling);
-        return Ok(());
-    }
+fn default_terminal_colors_str() -> clap::builder::OsStr {
+    is_colouring_supported().to_string().into()
+}
 
-    if env::args().any(|a| SUBMIT_ARG.contains(&a.as_str())) {
-        args.submit_genesis = true;
-        if let Ok(genesis_path) = env::var("IROHA2_GENESIS_PATH") {
-            args.genesis_path = Some(
-                ConfigPath::user_provided(&genesis_path)
-                    .wrap_err_with(|| "Required, because `--submit-genesis` was specified.")
-                    .wrap_err_with(|| format!("Could not read `{genesis_path}`"))?,
-            );
-        }
-    } else {
-        args.genesis_path = None;
-    }
+/// Iroha peer Command-Line Interface.
+#[derive(Parser, Debug)]
+#[command(name = "iroha", version = concat!("version=", env!("CARGO_PKG_VERSION"), " git_commit_sha=", env!("VERGEN_GIT_SHA")), author)]
+struct Args {
+    /// Path to the configuration file, defaults to `config.json`/`config.json5`
+    ///
+    /// Supported extensions are `.json` and `.json5`. By default, Iroha looks up for a
+    /// `config` file in the Current Working Directory with both supported extensions.
+    /// If the default config file is not found, Iroha will rely on default values and environment
+    /// variables. However, if the config path is set explicitly with this argument and the file
+    /// is not found, Iroha will exit with an error.
+    #[arg(
+        long,
+        short,
+        env("IROHA_CONFIG"),
+        value_name("PATH"),
+        value_parser(Path::user_provided_str),
+        value_hint(clap::ValueHint::FilePath)
+    )]
+    config: Option<Path>,
+    /// Whether to enable ANSI colored output or not
+    ///
+    /// By default, Iroha determines whether the terminal supports colors or not.
+    ///
+    /// In order to disable this flag explicitly, pass `--terminal-colors=false`.
+    #[arg(
+        long,
+        env,
+        default_missing_value("true"),
+        default_value(default_terminal_colors_str()),
+        action(clap::ArgAction::Set),
+        require_equals(true),
+        num_args(0..=1),
+    )]
+    terminal_colors: bool,
+    /// Whether the current peer should submit the genesis block or not
+    ///
+    /// The only one peer in the network should submit the genesis block.
+    ///
+    /// This argument must be set alongside with `genesis.file` and `genesis.private_key`
+    /// configuration options. If not, Iroha will exit with an error.
+    ///
+    /// This argument must be set if the amount of trusted peers in the config file
+    /// (`sumeragi.trusted_peers`) is less than 2, i.e. the network consists only from this peer
+    /// itself. Otherwise it would be impossible to receive genesis topology, and Iroha will exit
+    /// with an error.
```

Comment:
```suggestion
    /// This argument must be set if the amount of trusted peers in the config file
    /// (`sumeragi.trusted_peers`) is less than 2, i.e. the network consists only of this one peer.
    /// If the argument is not set, it will be impossible to receive genesis topology, and Iroha
    /// will exit with an error.
```

`to receive genesis topology` -- I don't really understand what it means

Reply:
> `to receive genesis topology` -- I don't really understand what it means

TBH, me neither! I copied the exact phrase from an error message. I will check it.